transform/convert_quadkey_and_Vertical_id_test.go

```diff
@@ -1254,3 +1719,88 @@ func TestConvertZToMinAltitudekey_11(t *testing.T) {
 		t.Fatal(result, error)
 	}
 }
+
+func TestConvertAltitudeKeyToMinMaxZ(t *testing.T) {
+	type args struct {
+		altitudekey          int64
+		altitudekeyZoomLevel int64
+		outputZoomLevel      int64
+		zBaseExponent        int64
+		zBaseOffset          int64
+	}
+	tests := []struct {
+		name    string
+		args    args
+		minZ    int64
+		maxZ    int64
+		wantErr bool
+	}{
+		{
+			name: "altitudekey must be offset converted",
+			args: args{
+				altitudekey:          0,
+				altitudekeyZoomLevel: 23,
+				outputZoomLevel:      23,
+				zBaseExponent:        23,
+				zBaseOffset:          8,
+			},
+			minZ:    -2,
+			maxZ:    -2,
+			wantErr: false,
+		},
+		{
+			name: "altitudekey must be zoom level converted",
+			args: args{
+				altitudekey:          0,
+				altitudekeyZoomLevel: 23,
+				outputZoomLevel:      22,
+				zBaseExponent:        25,
+				zBaseOffset:          -2,
+			},
+			minZ:    0,
+			maxZ:    0,
+			wantErr: false,
+		},
+		{
+			name: "minZ and maxZ must be different when zBaseOffset is not power of 2",
+			args: args{
+				altitudekey:          0,
+				altitudekeyZoomLevel: 23,
+				outputZoomLevel:      23,
+				zBaseExponent:        25,
+				zBaseOffset:          7,
+			},
+			minZ:    -2,
+			maxZ:    -1,
+			wantErr: false,
+		},
+		{
+			name: "minZ and maxZ must be different when outputZoom or altitudekeyZoomLevel is greater than zBaseExponent",
+			args: args{
+				altitudekey:          3,
+				altitudekeyZoomLevel: 26,
+				outputZoomLevel:      26,
+				zBaseExponent:        25,
+				zBaseOffset:          -2,
+			},
+			minZ:    6,
+			maxZ:    7,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotMinZ, gotMaxZ, err := ConvertAltitudeKeyToMinMaxZ(tt.args.altitudekey, tt.args.altitudekeyZoomLevel, tt.args.outputZoomLevel, tt.args.zBaseExponent, tt.args.zBaseOffset)
+			if (err != nil) != tt.wantErr {

```

Comment:
Should we just check if there is an error?
What about checking if error.Error() matches an expected value?
I don't like table-driven tests in the first place.
Because it requires counting breakpoints when you want to stop a particular test.
It also requires updating all rows when you want to add a new check.
If you end up doing t.Run(_) anyway, I think it's better to have separate tests for the first.
If you're extracting common code, it's better to create a common function.
Why do you bother replacing the tests with table-driven tests?

Reply:
1. It was following convention of Go tests
2. Adding a test case for every single function is adding trivial codes

Now it is changed to use single function as every test cases and added helper to resolve 2. at [ebe0d0e](https://github.com/trajectoryjp/spatial_id_go/pull/29/commits/ebe0d0ef9fe47fe323d9bd0c926f1e6cb0385d2c)

Reply:
I see.